rtos/Thread.cpp

```diff
@@ -238,10 +237,12 @@ uint32_t Thread::free_stack() {
     uint32_t size = 0;
     _mutex.lock();
 
+#if defined(TARGET_CORTEX_M)
```

Comment:
Why are we adding that? I would hope the new Cortex A support in RTX is not that different